### PR TITLE
Adding nCr_mod_p

### LINKED ISSUE #1205: nCr mod P in C and kotlin
Please assign it to me. @jainaman224

### PR DESCRIPTION
**Pull request by shubhammantri1**
Fixes #1205 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- nCr mod P in Kotlin

#### Languages Used:
- Kotlin

#### Files Added:
- nCr_mod_P.kt 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: nCr mod P/nCr_mod_P.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    println("Please Enter Three Integers.")
+    var n = scanner.nextInt()
+    var r = scanner.nextInt()
+    var mod = scanner.nextInt()
+    var ans = nCr(n, r, mod)
+    if (ans == -1) {
+        println("n can never be less than r.")
+    }
+    else {
+        println("The value of nCr($n,$r)%$mod = $ans")
```
Comment (singh-shreya6) [inline nCr mod P/nCr_mod_P.kt]: space around % would be nice.

File: nCr mod P/nCr_mod_P.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            k = r
+        }
+        for (j in k downTo 1) {
+            pascal[j] = (pascal[j] + pascal[j-1]) % mod
+        }
+    }
+    return pascal[r]
+}
+
+// Driver Function for nCr
```
Comment (singh-shreya6) [inline nCr mod P/nCr_mod_P.kt]: You can remove this comment.

File: nCr mod P/nCr_mod_P.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+fun nCr(n: Int, r: Int, mod: Int):Int {
+    if(n < r) {
+        return -1
+    }
+    var pascal = IntArray(r + 1)
+    pascal[0] = 1
+    for (i in 1 until r+1) {
+        pascal[i] = 0
+    }
+    // We use the known formula nCr = (n-1)C(r) + (n-1)C(r-1) for computing the values.
```
Comment (singh-shreya6) [inline nCr mod P/nCr_mod_P.kt]: A blank line before comment would be better.

Comment (shubhammantri1) [reply nCr mod P/nCr_mod_P.kt]: Done

File: nCr mod P/nCr_mod_P.kt (hunk 2 above)
Comment (shubhammantri1) [reply nCr mod P/nCr_mod_P.kt]: Done

File: nCr mod P/nCr_mod_P.kt (hunk 1 above)
Comment (shubhammantri1) [reply nCr mod P/nCr_mod_P.kt]: Done

File: nCr mod P/nCr_mod_P.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    for (i in 1 until n+1) {
+        var k: Int
+        if (i < r) {
+            k = i
+        }
+        else {
+            k = r
+        }
+        for (j in k downTo 1) {
+            pascal[j] = (pascal[j] + pascal[j-1]) % mod
```
Comment (singh-shreya6) [inline nCr mod P/nCr_mod_P.kt]: space around - in j-1

File: nCr mod P/nCr_mod_P.kt [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        return -1
+    }
+    var pascal = IntArray(r + 1)
+    pascal[0] = 1
+    for (i in 1 until r+1) {
+        pascal[i] = 0
+    }
+
+    // We use the known formula nCr = (n-1)C(r) + (n-1)C(r-1) for computing the values.
+    for (i in 1 until n+1) {
```
Comment (singh-shreya6) [inline nCr mod P/nCr_mod_P.kt]: space around +

File: nCr mod P/nCr_mod_P.kt [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import java.util.*
+
+// Function for nCr
+fun nCr(n: Int, r: Int, mod: Int):Int {
+    if(n < r) {
+        return -1
+    }
+    var pascal = IntArray(r + 1)
+    pascal[0] = 1
+    for (i in 1 until r+1) {
```
Comment (singh-shreya6) [inline nCr mod P/nCr_mod_P.kt]: space around +

Comment (shubhammantri1) [reply nCr mod P/nCr_mod_P.kt]: Done

File: nCr mod P/nCr_mod_P.kt (hunk 5 above)
Comment (shubhammantri1) [reply nCr mod P/nCr_mod_P.kt]: Done

File: nCr mod P/nCr_mod_P.kt (hunk 4 above)
Comment (shubhammantri1) [reply nCr mod P/nCr_mod_P.kt]: Done

**Comment by jainaman224** [comment]
@singh-shreya6 Please review.

**Comment by shubhammantri1** [comment]
Please Approve @jainaman224 !

**Pull request merged**